data/txHandler.go

```diff
@@ -514,6 +522,11 @@ func (handler *TxHandler) postProcessCheckedTxn(wi *txBacklogMsg) {
 	if wi.verificationErr != nil {
 		// disconnect from peer.
 		handler.postProcessReportErrors(wi.verificationErr)
+		// if in synchronous mode, signal the completion of the operation
+		if wi.syncCh != nil {
+			wi.syncCh <- network.Disconnect
+			return
+		}
 		logging.Base().Warnf("Received a malformed tx group %v: %v", wi.unverifiedTxGroup, wi.verificationErr)
```

Comment:
Should logging be above the syncCh handling so that it logs either way?

Reply:
good catch, moved up